admin/entrypoint.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# the goal of this script is to inject environment variables into the frontend
+# by creating a javascript file that will be injected in the boot file
+WWW_DIR=/app
+ENV_PATH="${WWW_DIR}/env.js"
+# Create the file
+echo -n "window.env = {" > "${ENV_PATH}"
+COMMA=""
+for envrow in $(printenv); do
+  IFS='=' read -r key value <<< "${envrow}"
+  case "${key}" in
+    "API_URL" | "API_PATH" | "AUTH_CLIENT_ID")
+      echo -n "${COMMA}" >> "${ENV_PATH}"
+      echo -n "${key}: '${value}'" >> "${ENV_PATH}"
+      COMMA=","
+      ;;
+    *)
+      # ignore the key
+      ;;
+  esac
+done
+echo "};" >> "${ENV_PATH}"
+
+# execute the web server
+nginx -g 'daemon off;'
```

Comment:
@ymarcon  

```sh
# execute the command passed to the entrypoin
# see https://stackoverflow.com/questions/39082768/what-is-the-difference-between-cmd-and-entrypoint-in-a-dockerfile

[ "$#" -eq 0 ] && nginx -g 'daemon off;' || exec "$@"

```

Reply:
Only useful if you want to override the start command. Why would we need that?

Reply:
The LLM explains it better than me :) That's your call at the end!

Overriding the start command in a Docker `ENTRYPOINT` script can make a container more flexible by enabling it to serve multiple purposes based on how it is executed. Let’s break down your example and why it’s useful:

### Understanding the Script
```bash
[ "$#" -eq 0 ] && nginx -g 'daemon off;' || exec "$@"
```

1. **Condition Explanation**:
   - `[ "$#" -eq 0 ]`: Checks if no arguments are passed when the container is run.
   - `&& nginx -g 'daemon off;'`: If no arguments are passed, it defaults to starting the `nginx` server in the foreground (`daemon off` is required for proper containerized behavior).
   - `|| exec "$@"`: If arguments are provided, they are treated as a command to execute.

2. **Purpose**:
   - If you run the container without arguments (e.g., `docker run my-nginx`), it starts `nginx` as the default service.
   - If you run the container with arguments (e.g., `docker run my-nginx echo Hello`), it executes the provided command (`echo Hello`) instead of starting `nginx`.

### Why Use This Pattern?

1. **Default Behavior with Flexibility**:
   - You want the container to have a sensible default (e.g., running `nginx`), but you also want the ability to override this behavior when needed.

2. **Ease of Debugging**:
   - In development or debugging scenarios, you might want to run an alternate command like `/bin/bash` or `cat /etc/nginx/nginx.conf` to inspect the container environment, without modifying the Dockerfile.

3. **Reusable Containers**:
   - This makes the container more reusable because it doesn't assume the only purpose is to run `nginx`. The same image can also be used for auxiliary tasks.

4. **Improved CI/CD Pipelines**:
   - In a CI/CD pipeline, you might want to run the container with different commands, such as testing the configuration (`nginx -t`) before deploying.

5. **Aligning with Best Practices**:
   - Docker best practices often suggest making images more generic and flexible, so they can handle both default behaviors and user-specified commands.

### Alternative Without This Logic
Without this script, you'd need to create separate Docker images or modify the `ENTRYPOINT` in every scenario:
- A fixed `ENTRYPOINT` like `CMD ["nginx", "-g", "daemon off;"]` would always run `nginx`, making it harder to debug or override.

### Practical Example
```bash
docker run my-nginx
# Runs nginx as intended.

docker run my-nginx /bin/bash
# Launches a shell for debugging.

docker run my-nginx nginx -t
# Tests nginx configuration.
```

This design pattern makes your Docker image more versatile and suitable for a broader range of use cases.